frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -2444,252 +2489,292 @@ ApplicabilityResult instantiateSignature(Context* context,
+    ? Resolver::createForInitializer(rc, fn, poiScope, inputSig, rr)
+    : Resolver::createForFunction(rc, fn, poiScope, inputSig, rr);
+
+  if (isInitializer) {
+    CHPL_ASSERT(visitor.initResolver.get());
+    auto qt = QualifiedType(QualifiedType::VAR, VoidType::get(context));
+    visitor.returnType = std::move(qt);
+  }
+
+  // Walk the body to resolve it.
+  if (fn->body()) fn->body()->traverse(visitor);
+
+  // There were outer variables we could not type, so give up.
+  if (!visitor.outerVariables.isEmpty() && rc->isEmpty()) {
+    return nullptr;
+  }
+
+  // then, compute the return type if it is not an initializer
+  if (!isInitializer) {
+    computeReturnType(visitor);
+
+  // else, potentially write out a new initializer signature
+  } else if (visitor.initResolver) {
+    finalSig = visitor.initResolver->finalize();
+  }
+
+  // then, resolve '=' and add any copy init/deinit calls as needed
+  callInitDeinit(visitor);
+
+  // then, handle return intent overloads and maybe-const formals
+  adjustReturnIntentOverloadsAndMaybeConstRefs(visitor);
+
+  // check that throws are handled or forwarded
+  // TODO: Call for initializers as well, and remove checks in the resolver.
+  if (!isInitializer) checkThrows(rc, rr, fn);
+
+  // TODO: can this be encapsulated in a method?
+  resolvedPoiInfo.swap(visitor.poiInfo);
+  resolvedPoiInfo.setResolved(true);
+  resolvedPoiInfo.setPoiScope(nullptr);
+
+  CHPL_ASSERT(inputSig == finalSig || isInitializer);
+
+  auto ret = toOwned(new ResolvedFunction(finalSig,
+                                  fn->returnIntent(),
+                                  std::move(rr),
+                                  std::move(resolvedPoiInfo),
+                                  std::move(visitor.returnType),
+                                  std::move(visitor.poiTraceToChild),
+                                  std::move(visitor.sigAndInfoToChildPtr)));
+  return ret;
 }
 
+// Forward declaration so that that query storage actions can be specialized.
 static const ResolvedFunction* const&
-resolveFunctionByInfoQuery(Context* context,
+resolveFunctionByInfoQuery(ResolutionContext* rc,
                            const TypedFnSignature* sig,
-                           PoiInfo poiInfo) {
-  QUERY_BEGIN(resolveFunctionByInfoQuery, context, sig, poiInfo);
+                           PoiInfo poiInfo);
 
-  const UntypedFnSignature* untypedSignature = sig->untyped();
-  const AstNode* ast = parsing::idToAst(context, untypedSignature->id());
-  const Function* fn = ast->toFunction();
+static const owned<ResolvedFunction>&
+resolveFunctionByPoisQuery(ResolutionContext* rc, PoiInfo::Trace poiTrace);
+
+// TODO: This can be used later to implement the "rewinding" necessary to
+// resolve an exterior call to a nested function. Right now it is not
+// clear if that will ever be possible outside of the API, so it remains
+// unimplemented (but this specialization will stay as a reminder).
+template <typename Args> struct
```

Comment:
Please update the above comment to describe what this does. Is it just doing the same as the default behavior? Is it disabling something?